internal/venice-common/src/main/java/com/linkedin/venice/acl/handler/AbstractStoreAclHandler.java

```diff
@@ -89,9 +156,28 @@ public void channelRead0(ChannelHandlerContext ctx, HttpRequest req) throws SSLP
       return;
     }
 
-    X509Certificate clientCert = extractClientCert(ctx);
     String method = req.method().name();
-    AccessResult accessResult = checkAccess(uri, clientCert, storeName, method);
+    VeniceConcurrentHashMap<AclKey, CachedAcl> storeAclCache =
+        originalChannel.attr(STORE_ACL_CHECK_RESULT_ATTRIBUTE_KEY).get();
+    if (storeAclCache == null) {
+      originalChannel.attr(STORE_ACL_CHECK_RESULT_ATTRIBUTE_KEY).setIfAbsent(new VeniceConcurrentHashMap<>());
+      storeAclCache = originalChannel.attr(STORE_ACL_CHECK_RESULT_ATTRIBUTE_KEY).get();
+    }
+    AclKey aclKey = new AclKey(storeName, method);
+
+    AccessResult accessResult = storeAclCache.compute(aclKey, (ignored, value) -> {
+      long currentTimestamp = time.getMilliseconds();
+      if (value == null || currentTimestamp - value.timestamp > cacheTTLMs) {
```

Comment:
So... if we set the `cacheTTLMs` to -1, then the code will always go into this branch which means the caching is effectively disabled... but it might be cleaner if there is a way to completely short-circuit the caching logic, rather than let it run in a way where every query evaluates to a cache miss...

Reply:
Sure, I will add a new config flag to completely disable this feature.